Workshop syllabus
=================

 
**Basic Bioinformatics for Genome Assembly and Evolutionary Genomics**

Nova Southeastern University Oceanographic Center
October 28 - November 15, 2019

SYLLABUS

Workshop Description
--------------------

This ten-day workshop will provide a general and hands-on instruction on the concepts, methods, and software used in genomics research for those who are beginners for bioinformatics in genomics research. Contents have been carefully selected for graduate students, postdocs, and researchers at different stages who are interested in designing genomics projects or already generating genome data to acquire the skills necessary to assemble and annotate genomes and transcriptomes, and to perform phylogenomic and population genomic analyses. The workshop provides introduction to the concepts, methods, and software used in genomics research. Participants review theory and methods and experience hands-on tutorials using the tools necessary for high-throughput sequencing data analysis. Participants learn how to choose which software packages best fit the data, how to use common software packages, and how to interpret the results. By the end of the workshop, participants are expected to be able to conduct genomic analyses and critically evaluate current genomics research literature.

Organizers
----------
| *Stephen O’Brien*, Ph.D., Professor, Halmos College of Natural Sciences and Oceanography, Nova Southeastern University
| *Jose Lopez*, Ph.D., Professor, Halmos College of Natural Sciences and Oceanography, Nova Southeastern University 
| *Mahmood Shivji*, Ph.D., Professor, Halmos College of Natural Sciences and Oceanography, Nova Southeastern University 

Instructors
-----------
| *Klaus-Peter Koepfli*, Ph.D., Conservation Biologist, Center for Species Survival, Smithsonian Conservation Biology Institute
| *Hao Meng*, Ph.D. Candidate, Center for Life Sciences, Peking University
| *Xin Sun*, Ph.D. Candidate, Center for Life Sciences, Peking University
| *Shu-Jin Luo*, Ph.D., Associate Professor, School of Life Sciences, Peking University

Workshop Goals
--------------
The purpose of the workshop is to provide participants with the knowledge and tools to be able to conduct their own genomic analyses.

Prerequisite for Participants
-----------------------------
Participants should have completed basic genetics and evolutionary biology coursework or have previous experience with genetics, genomics, and evolutionary biology.

 
Summary of Contents
-------------------
1.	**Overview** of whole genome sequencing and reduced representation methods
2.	**Hands-on training** in computational tools used in genomics research
3.	**WGS data processing**: Genome assembly and annotation; read mapping, alignment and variant calling
4.	**Gene mapping**: GWAS and linkage mapping
5.	**Phylogenomics**: Phylogeny reconstruction with genome data sets
6.	**Population genomics**: Genomic diversity, gene flow, and historical demography
7.	**Selection**: Detection of natural selection with population genomic data sets

Course Format
-------------
This workshop composes of lectures, class discussions and extensive computer work. Most days include a combination of formal lectures and analytical work using a computer.

Learning Outcomes
-----------------
By the end of this workshop, participants will: 
| 1.	Be able to perform analyses on whole genome data or portions thereof, including quality assessment, assembly, annotation, variant calling, estimation of diversity and inbreeding, detection of admixture, and demographic inference. 
| 2.	Understand the conceptual basis for choosing analysis parameters and software packages for conservation genomics research projects. 

Suggested Reading 
-----------------
The suggested references below will enhance course learning experience.

| Ekblom, R., J. B. W. Wolf. 2014. A field guide to whole‐genome sequencing, assembly and annotation. *Evolutionary Applications*. 7(9): 1026-42. DOI: 10.1111/eva.12178 
| Fuentes-Pardo, A.P., D.E. Ruzzante. 2017. Whole-genome sequencing approaches for conservation biology: advantages, limitations and practical recommendations. *Molecular Ecology* 26(20): 5368-5406. 
| Hendricks, S., E. C. Anderson, T. Antao, L. Bernatchez, B. R. Forester, B. Garner, B. K. Hand, P. A. Hohenlohe, M. Kardos, B. Koop, A. Sethuraman, R. S. Waples, G. Luikart 2018. Recent advances in conservation and population genomics data analysis. *Evolutionary Applications* | 11:1197–1211. 
| Sims, D., I. Sudbery, N. E. Ilott, A. Heger, C. P. 2014. Sequencing depth and coverage: key considerations in genomic analyses. *Nature Reviews Genetics* 15(2):121-32.
| Steiner, C. C., Putnam, A.S., Hoeck, P.E.A, and O. A. Ryder. 2013. Conservation genomics of threatened animal species. *Annual Review of Animal Biosciences* 1:261–281. 
| Yandell, M., D. Ence. 2012. A beginner's guide to eukaryotic genome annotation. *Nature Reviews Genetics* 13(5):329-42. 
| Yoder, A.D., J.W. Poelstra, G.P. Tiley, R.C. Williams. 2018. Neutral theory is the foundation of conservation genetics. *Molecular Biology and Evolution* 35(6): 1322-1326.

 
Dates 
-----
The workshop contains an 8-hour section per day, for 10 days, or two weeks with a one-week break in between.

-	Week 1: October 28 (Mon) - November 1 (Fri)
-	Week 2: November 11 (Mon) – 15 (Fri)

We will have a follow-up section with participants bringing in data, questions, and feedback, to make sure that the workshop has fulfilled its goals. This is tentatively scheduled sometime during November 20-22.

Daily Schedule
--------------
On each workshop day, lectures or course activities will begin at :9:00: am and continue until around 5:00 pm, with 1 hour (:12:30:-1:30 pm) for lunch. A 30-minute break will occur once each morning and afternoon. Lectures and tutorials would be given by Hao Meng, Xin Sun and Shu-Jin Luo, unless otherwise specified.

**Week 1**

Day 1: October 28 (Mon)	**Overview 1: Environment and software** 

We will introduce the Unix operating system, Python, and commercial software and pipelines. 

:9:00:	 	Introduction, discussion and course logistics 
:10:00:		**Tutorial**: Introduction to Cloud computing 
:10:30:		Break
:11:00:		**Tutorial**: Introduction to Unix 
:12:30:		Lunch break
:1:30:		**Tutorial**: Introduction to Python I (Elizaveta Shmakova)
:2:30:		**Tutorial**: Introduction to Python II 
:3:30:		Break
:4:00:		**Lecture**: Introduction to software and pipelines

Day 2: October 29 (Tue) 	**Genome assembly 1: Assembly principles and methods**

We will discuss concepts and methods behind de novo and reference-guided whole genome assembly, using examples from the published data. We will discuss methods for evaluating the quality of genome assemblies. Computational resources required for accurate and efficient assembly will also be discussed. We will assemble (1) a mitogenome, (2) a genome of a bacteria species, and (3) contigs from one short and one long scaffolds from an animal genome, with SOAPdenovo and/or Geneious. Day 3 will cover (1) and (2).

:9:00:	**Lecture**: Concepts and methods behind de novo and reference-guided whole genome assembly 
:10:30:		Break
:11:00:	**Tutorial**: Introduction to genome assembly with Geneious 
:12:30:		Lunch break
:1:30:		**Tutorial**: Mitogenome assembly with SOAPdenovo and/or others software
:3:00:		Break
:3:30:		**Tutorial**: Bacteria genome assembly 

Day 3 October 30 (Wed)	**Genome assembly 2** 

Continued from Day 3, we will assemble (3) contigs from one short and one long scaffolds from an animal genome, with SOAPdenovo. We will learn to evaluate the quality of genome assemblies.

:9:00:		**Tutorial**: Genome assembly of contigs from scaffolds
:10:30:		Break 
:11:00:		**Tutorial**: Genome assembly of contigs from scaffolds (cont.)
:12:30:		Lunch break
:1:30:		**Tutorial**: Genome assembly quality assessment
:3:00:		Break
:3:30:		**Lecture**: Moscow genomic workshop share (Elizaveta Shmakova)

Day 4 October 31 (Thu)	**Genome annotation 1** 

We will learn how to mask the repeat sequences. The repetitive sequences will be identified by a combination of homology searching (using RepeatMasker with the Repbase transposable element library) and ab initio prediction (using RepeatModeler). We will also use Tandem Repeats Finder to predict Tandem repeats. Finally, they are combined into the final repeat annotation. 

:9:00:		**Lecture**: Introduction to genome annotation
:10:30:		Break
:11:00:		**Tutorial**: Repetitive elements annotation	with ab initio prediction
:12:30:		Lunch break
:1:30:		**Tutorial**: Repetitive elements annotation	with homology searching
:3:00:		Break
:3:30:		**Tutorial**: Integration of different methods for repetitive elements annotation

Day 5 November 1 (Fri)	**Genome annotation 2** 

We will annotate the coding gene structure by integrating ab initio predictions, homology-based gene predictions and direct gene models produced by transcriptome assembly. We will conduct the transcriptome-based prediction with Trinity, Program to Assemble Spliced Alignment (PASA), Hisat2 and Stringtie. For de novo prediction, Augustus and SNAP will be used to predict coding regions in the repeat-masked genome. And gene models from the PASA dataset are used to create the input training file for AUGUSTUS as well as SNAP. For homology-based methods, Proteins sequences from several species will be mapped against the genome with Exonerate. Finally, all gene models derived from these 3 methods will be integrated by EvidenceModeler (EVM) into a nonredundant gene set with different weights. We then will filter short low-quality genes (encoding proteins with <50 amino acids) exhibiting premature termination. 

:9:00:		**Tutorial**: Coding region annotation-gene prediction
:10:30:		Break
:11:00:		**Tutorial**: Coding region annotation-homology-based
:12:30:		Lunch break
:1:30:		**Tutorial**: Coding region annotation- transcriptome-based
:3:00:		Break
:3:30:		**Tutorial**:  Integration of gene annotations with various methods 

**Week 2**

Day 6 November 11 (Mon)	**Overview 2: Study design and sequencing platforms** (Klaus Koepfli) 

In a series of lectures, we will survey important concepts used in conservation genomics. Using examples from the lecturer’s own research and the published scientific literature, students will learn how to conduct a conservation genomics project from the beginning to completion. Concepts to be discussed include population history, effective population size, sampling of individuals and genetic markers, next-generation sequencing approaches for sampling genetic variants (e.g., whole genome sequencing and reduced representation approaches), genome coverage, and estimating genetic diversity and inbreeding. In the afternoon, we will discuss for first-, second- and third-generation technologies used for whole genome sequencing, including those based on long-read and short-read methods. The advantages and limitations of each method will also be discussed.  We will then present an overview of the genome assembly pipeline used for the Vertebrate Genomes Project and the history and goals of this project.

:9:00:	**Lecture**: Concepts and study design in population genomics	
:10:30:		Break
:11:00:	**Lecture**: Overview of sequencing platforms 
:12:30:		Lunch break
:1:30:		**Lecture**: Concepts and methods of whole genome assembly 
:3:00:		Break
:3:30:	**Lecture**: The Vertebrate Genomes Project and Earth BioGenome Project 

Day 7 November 12 (Tue)	**Read mapping and variant calling** 

We will map short reads to reference genome with BWA and samtools and call the variants with GATK, and filter the VCF files with VCFtools and BCFtools.

:9:00:		**Lecture**: Introduction to read mapping and variant calling
:10:30:		Break
:11:00:		**Tutorial**: Reads mapping with bwa and samtools
:12:30:		Lunch break
:1:30:		**Tutorial**: SNP calling with GATK
:3:00:		Break
:3:30:		**Tutorial**: Quality control of SNP calling 


Day 8 November 13 (Wed)	**Phylogenomics and coalescence**

Genome data (SNPs and single-copy gene sets across a set of taxa) will be used for phylogenomic inference with PAUP and RAxML. Species tree methods (e.g., ASTRAL) and MCMC divergence dating methods (e.g. BEAST and MrBayes) will be introduced. BUSCO assessments will be demonstrated for identification of single-copy orthologs.

:9:00:		**Lecture**: Introduction to phylogenetics and phylogenomics 
:10:30:		Break
:11:00:	**Tutorial**: Assemble SNPs dataset and single-copy orthologs for phylogenomic analysis 
:12:30:		Lunch break
:1:30:		**Tutorial**: Phylogenomic inference methods 
:3:00:		Break
:3:30:	**Tutorial**: Species tree methods and MCMC divergence dating methods 

Day 9 November 14 (Thu)	**Population Genomics 1: GWAS, Linkage mapping & QTL**

We will introduce the GWAS and learn how to use PLINK. We will learn the basic concepts of linkage mapping and give an example with MERLIN, and briefly introduce the QTL.

:9:00:		**Lecture**: Basis of gene mapping	
:10:30:		Break
:11:00:		**Tutorial**: GWAS with Plink
:12:30:		Lunch break
:1:30:		**Tutorial**: Linkage mapping with MERLIN
:3:00:		Break
:3:30:		**Tutorial**: Brief introduction to QTL

Day 10 November 15 (Fri)	**Population Genomics 2: Gene flow and selection**

We will analyze the population structure with ADMIXTURE, and conduct the PCA with smartpca in EIGENSOFT. We will use ADMIXTOOLS to detect the geneflow between populations, use G-PhoCS and PSMC to estimate the divergence time and demographic history. We will calculate several genetic parameters of the populations with VCFtools and visualize the results with R, to locate the region potentially under selection.

:9:00:	**Lecture**: Neutral theory-based population genetic analysis vs. detection of selection 
:10:30:		Break
:11:00:		**Tutorial**: Admixture, PCA and PSMC 
:12:30:		Lunch break
:1:30:		**Tutorial**: Gene flow detection 
:3:00:		Break
:3:30:		**Tutorial**: Selection detection 

